#2758 Fixed xid validation error handling

Patrykb0802 opened the pull request:
Issue: #2758 

Fixed xid validation for:

- Maintenance events;
- Scheduled events;
- Event handlers;
- Compound event detectors;
- Point links;
- Mailing lists;
- Publishers;

 by adding another condition or removing context messages when conditions are fulfilled

Patrykb0802 committed:
Fixed xid validation for:

- Maintenance events;
- Scheduled events;
- Event handlers;
- Compound event detectors;
- Point links;
- Mailing lists;
- Publishers;
by adding another condition or removing context messages when conditions are fulfilled

Limraj commented on src/com/serotonin/mango/web/dwr/CompoundEventsDwr.java:
```diff
@@ -122,6 +122,9 @@ public DwrResponseI18n saveCompoundEvent(int id, String xid, String name, int al
             response.addContextualMessage("xid", "validate.required");
         else if (!compoundEventDetectorDao.isXidUnique(xid, id))
             response.addContextualMessage("xid", "validate.xidUsed");
+        else if (StringUtils.isLengthGreaterThan(xid, 50)){
```
It is worth using the general xid validation method from XidUtils class.

Patrykb0802 committed:
Changed StringUtils xid validation to XidUtils.validateXid so now most components use dedicated xid validation

Limraj commented on src/com/serotonin/mango/web/dwr/EventHandlersDwr.java:
```diff
@@ -330,6 +332,9 @@ private DwrResponseI18n save(int eventSourceId, int eventTypeRef1,
 
 		DwrResponseI18n response = new DwrResponseI18n();
+
+		XidUtils.validateXid(response, eventService::isXidUnique, xid, handlerId);
```
1. This method should be used in the validate method on EventHandlerVO class; (next line "vo.validate(response)" )
2. The non-xid value should be validated, because then the mechanism that automatically generates xid will not work if the user left the field empty.

Patrykb0802 committed:
#2758 Fixed xid validation error handling

Moved xid validations to validate() method for each component:
- Compound Event
- Event Handler
- Mailing List
- Maintenance Event
- Point Link
- Scheduled Event

Patrykb0802 committed:
#2758 Fixed xid validation error handling

Removed unused imports left from previous commits

Patrykb0802 committed:
#2758 Fixed xid validation error handling

- Using xid and id instead of getXid() and getId() in validateXid() methods

Limraj reviewed (approved):
I did xid validation tests for the types:
- Maintenance events;
- Scheduled events;
- Event handlers;
- Compound event detectors;
- Point links;
- Mailing lists;
- Publishers;
